discord/ext/commands/core.py

```diff
@@ -1396,8 +1396,9 @@ def predicate(ctx):
         if not isinstance(ctx.channel, discord.abc.GuildChannel):
             raise NoPrivateMessage()
 
-        getter = functools.partial(discord.utils.get, ctx.author.roles)
-        if any(getter(id=item) is not None if isinstance(item, int) else getter(name=item) is not None for item in items):
+        author = ctx.author
+        getter = functools.partial(discord.utils.get, author.roles)
```

Comment:
And if you chose to remove that line just change this too.
```suggestion
        getter = functools.partial(discord.utils.get, ctx.author.roles)
```